ocaml/lambda/translcore.ml

```diff
@@ -1343,7 +1348,9 @@ and transl_function ~scopes e alloc_mode param arg_mode arg_sort return_sort
   let attr = default_function_attribute in
   let loc = of_location ~scopes e.exp_loc in
   let body = if region then maybe_region_layout return body else body in
-  let lam = lfunction ~kind ~params ~return ~body ~attr ~loc ~mode ~region in
+  let may_locally_allocate_in_parent = may_allocate_in_region body in
```

Comment:
This feels a bit wasteful in the case where the function has a region and does no local allocations: we end up looking at all it's allocations twice. Is there a simple way to only do one traversal here?